legacy.js

```diff
@@ -87,23 +88,14 @@ module.exports = function (layered) {
     },
     stream: function () {
       log_legacy3()
-      // HACK (mix) - I'm pretty sure this is a stream which only delivers one value,
-      // no updates. I've added a check to make sure it doesn't pushing something empty
+      return pCont(function (cb) {
+        layered.onReady(function () {
+          var source = streamNotify.listen()
+          source.push(mapGraph(layered.getGraph(), toLegacyValue))
 
-      var source = streamNotify.listen()
-      var legacyGraph
-
-      function pushGraph () {
-        legacyGraph = mapGraph(layered.getGraph(), toLegacyValue)
-
-        if (!isEmpty(legacyGraph)) source.push(legacyGraph)
-        else {
-          console.log('ssb-friends#stream : bummer race condition, trying again!')
-          setTimeout(pushGraph, 2e3)
-        }
-      }
-      pushGraph()
-      return source
+          cb(null, source)
+        })
+      })
```

Comment:
pull-cont + pull-notify seems like an odd combination, could this just be : 

```js
var source = streamNotify.listen()
layered.onReady(function () {
  source.push(mapGraph(layered.getGraph(), toLegacyValue))
})
return source
```

I guess the advantage of what you've written nothing can get pushed through streamNotify (if that were possible here) before onReady resolves.

Reply:
oh yeah, that works! good spotting!